src/main/java/filter/expression/Qualifier.java

```diff
@@ -136,28 +134,96 @@ public static boolean process(IModel model, FilterExpression expr, TurboIssue is
         return exprWithNormalQualifiers.isSatisfiedBy(model, issue, new MetaQualifierInfo(metaQualifiers));
     }
 
-    private static Optional<Integer> getCurrentMilestoneIndex(List<TurboMilestone> allMilestones) {
-        if (allMilestones.isEmpty()) {
-            return Optional.empty();
+    /**
+     * Get all milestones that can be aliased with current-[n] to current+[n]
+     *
+     * A milestone can be aliased if and only if it is open and has due date or it does not have due date but it is
```

Comment:
How about:
`Get all milestones which milestone aliases (current+/-[n]) can resolve to. This will henceforth be called aliasable milestones.`
`Aliasable milestones must have a due date. The only exception is if there is only one open milestone in the repo; it will be included in aliasable milestones regardless of its due date status.`